core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -54,6 +56,28 @@ public String getName()
         Session session = stateMachine.getSession();
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTableName());
 
+        // those two guards are here because some connectors violate getTableHandle contract for views or materialized views.
```

Comment:
```suggestion
        // those two guards are here because some connectors violate SPI contract and return views or materialized views from getTableHandle.
```